app/org/sagebionetworks/bridge/play/controllers/AuthenticationController.java

```diff
@@ -100,6 +102,7 @@ private Result signInWithRetry(final int retryCounter) throws Exception {
         final Study study = getStudyOrThrowException(json);
         try {
             session = authenticationService.signIn(study, signIn);
+            verifySupportedVersionOrThrowException(study, session);
```

Comment:
I think we should check this after line 95 as well. I don't know if this can happen, but if an application can be updated without losing its session, it would get back the session from Redis and we'd still want to block it at that time.